src/storage-types/src/shim.proto

```diff
@@ -0,0 +1,20 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+package mz_storage_types.shim;
+
+// This is just a shim message to try and get prost to generate a reference to
+// this shim as part of the `prost::Message` implementation for the
+// ProtoUpsertError type.
+//
+// The actual definition of `ProtoUpsertValueErrorShim` and its implementation
+// of `prost::Message` happens manually in `shim.rs`
+message ProtoUpsertValueErrorShim { }
```

Comment:
You need to document what `prost` command was used to generate the original version of `shim.rs`

Reply:
Hm, `shim.rs` was not created based on some original `prost` generated file. I just wrote a new rust module containing the `ProtoUpsertValueErrorShim` type and implemented the traits